AmplifyPlugins/API/APICategoryPlugin.xcodeproj/project.pbxproj

```diff
@@ -1351,9 +1397,15 @@
 			isa = PBXResourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				21598CE3239FF54E00529F29 /* RESTWithIAMIntegrationTests-amplifyconfiguration.json in Resources */,
 				B478F6E32374E0CF00C4F92B /* Main.storyboard in Resources */,
 				B478F6E12374E0CF00C4F92B /* Assets.xcassets in Resources */,
+				21598CF223A0164A00529F29 /* GraphQLWithUserPoolIntegrationTests-amplifyconfiguration.json in Resources */,
+				21F40A2B23A0423C0074678E /* GraphQLSyncBasedTests-amplifyconfiguration.json in Resources */,
+				21F40A2E23A0707E0074678E /* GraphQLModelBasedTests-amplifyconfiguration.json in Resources */,
+				21598CF323A0164A00529F29 /* GraphQLWithUserPoolIntegrationTests-awsconfiguration.json in Resources */,
```

Comment:
These are for different targets right? Can we not just have individual `amplifyconfiguration.json` files in the directory hierarchy for the specific target?

Reply:
the individual files are added to the HostApp's bundle and so individual targets load that file from Bundle and then passes it to AWSInfo. What i saw were individual targets are needed when we want to load more than one awsconfiguration since there isn't a clean way to reset things after calling `AWSInfo.configureDefaultAWSInfo`. This is why there are individual targets to testing REST IAM, REST UserPool, GraphQL IAM, GraphQL UserPool. since they all use different awsconfiguration.json files. the "Functional" tests rely on `amplifyconfiguration.json` and can be reset with Amplify.reset so we have the same target with many tests which uses the API key auth. 

the hierarchy would have to exist at the Copy To Bundle Resources of the HostApp since they are copied to the HostApp and not the target, as well as an hierarchy for S3 for the circleCI tests, so i opted to have different prefixes instead